### PR TITLE
Introduced rotatable panels for ContextualBalloon.

### PR DESCRIPTION
**Pull request by oskarwrobel**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduced rotatable panels for `ContextualBalloon`. Closes https://github.com/ckeditor/ckeditor5-ui/issues/491.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

Requires https://github.com/ckeditor/ckeditor5-theme-lark/pull/227

### CODE REVIEW

File: src/panel/balloon/contextualballoon.js [hunk 1]
```diff
@@ -235,3 +383,85 @@ export default class ContextualBalloon extends Plugin {
...
+		//
+		// @type {module:ui/button/buttonview~ButtonView}
+		this.buttonPrevView = this._createButtonView( locale.t( 'Prev' ) );
+
+		// Navigation button to switches panel to the previous one.
+		//
+		// @type {module:ui/button/buttonview~ButtonView}
+		this.buttonNextView = this._createButtonView( locale.t( 'Next' ) );
+
+		// Collection of the child views which creates balloon panel contents.
```
Comment (pjasiun) [inline src/panel/balloon/contextualballoon.js]:
```suggestion
		// Collection of the child views which creates rotator panel contents.
```
Copy -> Paste -> Read -> Update

File: src/panel/balloon/contextualballoon.js [hunk 2]
```diff
@@ -38,14 +41,8 @@ export default class ContextualBalloon extends Plugin {
...
 	 */
-	init() {
-		/**
-		 * The common balloon panel view.
-		 *
-		 * @readonly
-		 * @member {module:ui/panel/balloon/balloonpanelview~BalloonPanelView} #view
-		 */
-		this.view = new BalloonPanelView();
+	constructor( editor ) {
```
Comment (pjasiun) [inline src/panel/balloon/contextualballoon.js]: In the private talk, you told me that now the structure is:
```
BalloonPanelView => RotatorView => PanelView
```
I think this is useful information and you should leave it somewhere in a comment.

File: src/panel/balloon/contextualballoon.js [hunk 3]
```diff
@@ -71,47 +74,89 @@ export default class ContextualBalloon extends Plugin {
 		};
 
 		/**
-		 * Stack of the views injected into the balloon. Last one in the stack is displayed
-		 * as a content of {@link module:ui/panel/balloon/contextualballoon~ContextualBalloon#view}.
+		 * The currently visible view or `null` when there are no
+		 * views in the currently visible panel stack.
```
Comment (pjasiun) [inline src/panel/balloon/contextualballoon.js]: Is it possible to have no views in the currently visible panel stack?

Comment (oskarwrobel) [reply src/panel/balloon/contextualballoon.js]: You can check this property when ContextualBalloon is hidden. Is null then.

File: src/panel/balloon/contextualballoon.js [hunk 4]
```diff
@@ -235,3 +412,140 @@ export default class ContextualBalloon extends Plugin {
...
+	}
+
+	// @inheritDoc
+	render() {
+		super.render();
+
+		this.focusTracker.add( this.element );
+	}
+
+	// @returns {Boolean}
```
Comment (oskarwrobel) [inline src/panel/balloon/contextualballoon.js]: Incorrect comment. This function does not return anything.

File: theme/components/panel/balloonrotator.css [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+.ck .ck-balloon-rotator__navigation {
+	display: flex;
+	align-items: center;
+	justify-content: center;
+}
```
Comment (pjasiun) [inline theme/components/panel/balloonrotator.css]: Part of the CSS is in [ckeditor5-theme-lark repository](https://github.com/ckeditor/ckeditor5-theme-lark/compare/t/ckeditor5-ui/491?expand=1) and part is here. I need to say I have no idea how this separation works. Can anybody check if it is correct?

Comment (oskarwrobel) [reply theme/components/panel/balloonrotator.css]: Here we keep only scaffolding styles.

Comment (dkonopka) [reply theme/components/panel/balloonrotator.css]: Styles responsible for wireframes should land in the feature source, in theme-lark we are using backgrounds, colors etc. See: 
https://github.com/ckeditor/ckeditor5-link/blob/master/theme/linkform.css
https://github.com/ckeditor/ckeditor5-theme-lark/blob/master/theme/ckeditor5-link/linkform.css

File: src/panel/balloon/contextualballoon.js (hunk 4 above)
Comment (pjasiun) [reply src/panel/balloon/contextualballoon.js]: Then it is also incorrect function name :)

Comment (pjasiun) [reply src/panel/balloon/contextualballoon.js]: It should be `updateIsNarrow` in this case.

File: theme/components/panel/balloonrotator.css (hunk 5 above)
Comment (pjasiun) [reply theme/components/panel/balloonrotator.css]: So it is fine, right?

Comment (oskarwrobel) [reply theme/components/panel/balloonrotator.css]: Yes, it is.

File: src/panel/balloon/contextualballoon.js [hunk 6]
```diff
@@ -71,47 +74,89 @@ export default class ContextualBalloon extends Plugin {
...
-		const item = this._stack.get( this.view.content.get( 0 ) );
+		/**
+		 * A total number of all panels in the balloon.
+		 *
+		 * @private
+		 * @readonly
+		 * @observable
+		 * @member {Number} #_panelsLength
+		 */
+		this.set( '_panelsLength', 0 );
```
Comment (pjasiun) [inline src/panel/balloon/contextualballoon.js]: `_numberOfStacks`?

**Pull request merged**